### PR TITLE
Add IT for replays

### PR DESCRIPTION
**Pull request by skiddykong**

### CODE REVIEW

File: framework-tools-test/framework-tools-it/pom.xml [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+        <artifactId>framework-tools-test</artifactId>
+        <groupId>uk.gov.justice</groupId>
+        <version>2.1.0-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>framework-tools-it</artifactId>
+
+    <dependencies>
+        <!--<dependency>-->
```
Comment (MartinYSpasov) [inline framework-tools-test/framework-tools-it/pom.xml]: Remove dead code.

File: framework-tools-test/framework-tools-it/src/test/java/uk/gov/justice/framework/tools/replay/TestEventLogRepository.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import org.apache.commons.dbcp2.BasicDataSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Standalone repository class to access event streams. To be used in integration testing
+ */
+public class TestEventLogRepository extends EventLogJdbcRepository {
+    private static final Logger LOGGER = LoggerFactory.getLogger(uk.gov.justice.services.test.utils.core.eventsource.TestEventLogRepository.class);
+    static final String SQL_FIND_ALL = "SELECT * FROM event_log";
```
Comment (MartinYSpasov) [inline framework-tools-test/framework-tools-it/src/test/java/uk/gov/justice/framework/tools/replay/TestEventLogRepository.java]: SQL_FIND_ALL is unused, can you please remove it.

File: framework-tools-replay/start.sh [hunk 3]
```diff
@@ -12,3 +12,11 @@ java -jar target/event-buffer-liquibase-0.33.0.jar --url=jdbc:postgresql://local
 
 # Replay event streams
 java -jar target/framework-tools-replay-1.0.0-SNAPSHOT-swarm.jar replay -c standalone-ds.xml -l notification-event-listener-1.0.12-SNAPSHOT.war
+
+
+
+
+
+
+java -jar  /Users/justin/code/framework-tools/framework-tools-test/framework-tools-it/target/test-classes/framework-tools-replay-2.1.0-SNAPSHOT-swarm.jar replay -c /Users/justin/code/framework-tools/framework-tools-test/framework-tools-it/target/test-classes/standalone-ds.xml -l /Users/justin/code/framework-tools/framework-tools-test/framework-tools-it/target/test-classes/framework-tools-test-listener-1.0.0.war
```
Comment (purple52) [inline framework-tools-replay/start.sh]: Is this used? Maybe this file shouldn't have changed

File: framework-tools-test/framework-tools-it/pom.xml [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+            <artifactId>event-buffer-liquibase</artifactId>
+            <version>${framework.version}</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>uk.gov.justice.utils</groupId>
+            <artifactId>test-utils-logging-simple</artifactId>
+            <type>pom</type>
+            <scope>test</scope>
+            <version>1.5.0</version>
```
Comment (purple52) [inline framework-tools-test/framework-tools-it/pom.xml]: Hardcoded version number? Is this needed?

File: framework-tools-test/framework-tools-it/pom.xml [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+                        <goals>
+                            <goal>stop</goal>
+                        </goals>
+                    </execution>
+                </executions>
+                <dependencies>
+                    <dependency>
+                        <groupId>com.h2database</groupId>
+                        <artifactId>h2</artifactId>
+                        <version>1.4.187</version>
```
Comment (purple52) [inline framework-tools-test/framework-tools-it/pom.xml]: Should be handled by a property in the top level pom

File: framework-tools-test/framework-tools-it/src/test/java/uk/gov/justice/framework/tools/replay/ReplayIntegrationIT.java [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+    public boolean runCommand(final String command) throws Exception {
+
+        final Process exec = Runtime.getRuntime().exec(command);
+        final BufferedReader reader =
+                new BufferedReader(new InputStreamReader(exec.getInputStream()));
+
+        boolean matches = false;
+        String line = "";
+        while ((line = reader.readLine()) != null) {
+            Pattern p = Pattern.compile(".*caught a fish.*", Pattern.MULTILINE | Pattern.DOTALL);
```
Comment (purple52) [inline framework-tools-test/framework-tools-it/src/test/java/uk/gov/justice/framework/tools/replay/ReplayIntegrationIT.java]: Pattern could be moved outside the loop and made final

File: framework-tools-test/framework-tools-it/src/test/java/uk/gov/justice/framework/tools/replay/TestEventLogRepository.java [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+        this.datasource = dataSource;
+        this.logger = LOGGER;
+
+    }
+
+    public TestEventLogRepository(final String contextName) throws SQLException {
+        this(jdbcUrlFrom(contextName), contextName, contextName, getDriver(jdbcUrlFrom(contextName)).getClass().getName());
+    }
+
+    public static TestEventLogRepository forContext(final String contextName) {
```
Comment (MartinYSpasov) [inline framework-tools-test/framework-tools-it/src/test/java/uk/gov/justice/framework/tools/replay/TestEventLogRepository.java]: The method isn't used.

File: framework-tools-test/framework-tools-it/src/test/java/uk/gov/justice/framework/tools/replay/TestEventLogRepository.java [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+ */
+public class TestEventLogRepository extends EventLogJdbcRepository {
+    private static final Logger LOGGER = LoggerFactory.getLogger(uk.gov.justice.services.test.utils.core.eventsource.TestEventLogRepository.class);
+    static final String SQL_FIND_ALL = "SELECT * FROM event_log";
+
+    private final DataSource datasource;
+
+    public TestEventLogRepository(final DataSource datasource) {
+        this.datasource = datasource;
+        this.logger = LOGGER;
```
Comment (MartinYSpasov) [inline framework-tools-test/framework-tools-it/src/test/java/uk/gov/justice/framework/tools/replay/TestEventLogRepository.java]: Do we need the logger?

File: framework-tools-test/framework-tools-it/src/test/java/uk/gov/justice/framework/tools/replay/TestProperties.java [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+/**
+ * Loads test.properties files from classpath and makes property values available through an
+ * accessor method
+ */
+public class TestProperties {
+    private static TestProperties instance;
```
Comment (MartinYSpasov) [inline framework-tools-test/framework-tools-it/src/test/java/uk/gov/justice/framework/tools/replay/TestProperties.java]: Remove unused variable.

File: framework-tools-test/framework-tools-it/src/test/java/uk/gov/justice/framework/tools/replay/ReplayIntegrationIT.java [hunk 10]
```diff
@@ -0,0 +1,176 @@
...
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ReplayIntegrationIT {
+
+    private static final String SWARM_DEBUG_MODE = "-Xdebug -Xrunjdwp:transport=dt_socket,server=y,suspend=y,address=5005";
+
+    private static final TestProperties TEST_PROPERTIES = new TestProperties("test.properties");
+    private static final String H2_DRIVER = "org.h2.Driver";
```
Comment (MartinYSpasov) [inline framework-tools-test/framework-tools-it/src/test/java/uk/gov/justice/framework/tools/replay/ReplayIntegrationIT.java]: We could move H2_DRIVER and SWARM_DEBUG_MODE constants into test.properties.

File: framework-tools-test/framework-event-listener/pom.xml [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>uk.gov.justice.framework.tools</groupId>
+    <artifactId>framework-tools-test-listener</artifactId>
+    <version>1.0.0</version>
+
+    <packaging>war</packaging>
+
+    <properties>
+        <cpp.service-component>EVENT_LISTENER</cpp.service-component>
```
Comment (DesMarshall01) [inline framework-tools-test/framework-event-listener/pom.xml]: Is this used ?

File: framework-tools-test/framework-tools-it/src/test/java/uk/gov/justice/framework/tools/replay/ReplayIntegrationIT.java [hunk 12]
```diff
@@ -0,0 +1,176 @@
...
+    }
+
+
+    private String createCommandToExecuteReplay() {
+
+        final String replayJarLocation = getResource("framework-tools-replay*.jar");
+        final String standaloneDSLocation = getResource("standalone-ds.xml");
+        final String listenerLocation = getResource("framework-tools-test-listener*.war");
+
+        String deubug = "";
```
Comment (DesMarshall01) [inline framework-tools-test/framework-tools-it/src/test/java/uk/gov/justice/framework/tools/replay/ReplayIntegrationIT.java]: Typo ?

File: framework-tools-test/framework-tools-it/src/test/java/uk/gov/justice/framework/tools/replay/ReplayIntegrationIT.java [hunk 13]
```diff
@@ -0,0 +1,176 @@
...
+            Pattern p = Pattern.compile(".*caught a fish.*", Pattern.MULTILINE | Pattern.DOTALL);
+            if (p.matcher(line).matches()) {
+                matches = true;
+            }
+            System.out.println(line);
+        }
+        return matches;
+    }
+
+    private void insertEventLogData(final UUID streamId) throws SQLException, InvalidSequenceIdException {
```
Comment (DesMarshall01) [inline framework-tools-test/framework-tools-it/src/test/java/uk/gov/justice/framework/tools/replay/ReplayIntegrationIT.java]: Parameter 'streamId' isn't used, should be removed

File: framework-tools-test/framework-tools-it/src/test/java/uk/gov/justice/framework/tools/replay/ReplayIntegrationIT.java [hunk 14]
```diff
@@ -0,0 +1,176 @@
...
+    @Before
+    public void setUpDB() throws Exception {
+        EVENT_LOG_REPOSITORY = new TestEventLogRepository(initEventStoreDb());
+    }
+
+    @Test
+    public void runReplayTool() throws Exception {
+        final String command = createCommandToExecuteReplay();
+
+        final UUID streamId = randomUUID();
```
Comment (DesMarshall01) [inline framework-tools-test/framework-tools-it/src/test/java/uk/gov/justice/framework/tools/replay/ReplayIntegrationIT.java]: streamId isn't used as insertEventLogData() ignores it, should be removed

File: framework-tools-test/framework-tools-it/src/test/java/uk/gov/justice/framework/tools/replay/ReplayIntegrationIT.java [hunk 15]
```diff
@@ -0,0 +1,176 @@
...
+        boolean matches = false;
+        String line = "";
+        while ((line = reader.readLine()) != null) {
+            Pattern p = Pattern.compile(".*caught a fish.*", Pattern.MULTILINE | Pattern.DOTALL);
+            if (p.matcher(line).matches()) {
+                matches = true;
+            }
+            System.out.println(line);
+        }
+        return matches;
```
Comment (DesMarshall01) [inline framework-tools-test/framework-tools-it/src/test/java/uk/gov/justice/framework/tools/replay/ReplayIntegrationIT.java]: Should really kill process or better, check that it completes at some point

**Pull request merged**